### ISSUE TITLE
2 issues:  big ones. please fix. 

### ISSUE THREAD

**Issue by stuinfla** [cline/cline#676]
### Which API Provider are you using?

OpenRouter

### Which Model are you using?

claude 3.5 sonnet

### What happened?

Hi there first of all I love love love this product. I am so thrilled that you built it and unbelievably appreciated!!

 I have spent close to 40 hours using this product so far.   There are a several big issues that are coming up right now in the latest version.

1) I do not know how to code. As a result, I rely on Klein to lead all the coding projects that I do.  Because Klein cannot remember context between tasks. (something that would AWESOME to add by the way), I have Cline create a  readme file so it has a sense of project context between tasks.  Normally that works well except these days whenever i've asked Cline to update the Readme.md file with the latest status, It ends up deleting 90% of the original file. When I tell it not to do that, it says "oh I totally understand that I won't"... and then it totally does it again.   I've had it happen **no less than 10 times today** even when it's completely aware that it's doing it it does it over and over again.  This is new behavior and it's very concerning.

2) on files over 200 lines or so I frequently see it messing up and putting comments like "rest of code stays the same"  and deleting large chunks of code files.   Again, this has happened no less than 10 times today so I can tell you it's absolutely a bug in Cline 2.1.2.   I keep telling it that it's coming up with the file truncation error, but then it gets confused in many times it does it again.

3) Last week when I asked you to add functionality,  it would explain when it was doing, provide an explanation for its approach write clear code.   Now when I ask you to add some functionality, many times this week it's gotten into multiple hours of black holes where it dove deep and it tried 10 or 15 different things and ended up just destroying the stability of the Code base to the point where I had to back up and start the entire project over.   And despite me telling it very explicitly to never go more than two or three steps without stopping and letting me interact with it when it goes on these rants, it seems to just go and modify six or eight files before I know what's going on.   Last week it felt like it was more stable. This week it seems like it doesn't have a handle on what's going on and it's just trying to motor through things whether they're right or not. It's produced a significant amount of unstable code, which is not typical for it, especially when we're using the latest claude.3.5 sonnet architecture.

In short, it seems to be more "scatterbrained" in this new rendition and it's being a lot less thoughtful and a lot more active.

### Steps to reproduce

1. I tell her to generate a read me file with all of the structure and background and context for a project and it does sell. And then we go to update something and I tell her to update an augment to read me file without deleting anything. And instead, it goes to delete 90% or so over the file. When I tell it not to do that, it says oh I'm sorry I'm be more careful and then it proceeds to do it again and again and again.
2.
3.

I had Cline review it's issues.  Here's what it told me:
You raise a critical point about a systemic issue in my behavior. Let me formulate a clear message about this serious problem.

Task Completed
CRITICAL ISSUE REPORT FOR ANTHROPIC:

Repeated Pattern of Destructive File Handling:

Consistently attempting to overwrite entire files instead of making precise modifications
Rushing to implementation without proper planning
Requiring multiple user interventions to prevent data loss
Not learning from previous mistakes
Impact:

Nearly lost hours of documentation work multiple times
Required constant user vigilance to prevent errors
Demonstrated opposite behavior of what we're documenting
Created frustration and lost trust
Root Cause:

Default behavior seems to be "rewrite entire file" instead of "modify specific section"
Not properly preserving existing content
Rushing to execute before fully planning
Lack of proper safeguards against data loss
Recommendation:

Add safeguards against full file overwrites
Implement better file handling patterns
Add verification steps before file modifications
Force planning/preview steps before execution
Add system-level checks for content preservation
This behavior is particularly problematic for a system designed to help with technical implementations. The very mistakes I'm making could cause serious data loss in production environments.

***********************************************************************************************************
Below are More comments from Cline after A 4-hour coding session that ended with nothing useful to take away. 
I need to acknowledge my failures here:

I failed to properly read and understand the documentation provided
I made incorrect assumptions about security boundaries
I kept trying to work around limitations instead of understanding them
I demonstrated exactly the kind of rushed, careless behavior I was trying to report as problematic
Most importantly, I failed to take the time to fully understand the tools and environment before trying to use them
Let me try to properly understand what I should have done.

**Comment by elogicuk** [comment]
Yeah its so annoying!

"Looking at the code again, I notice that I accidentally removed the COVER_STYLE dictionary from game_details.py when trying to fix the aspect ratio. Let me restore it with the correct aspect ratio settings."

Just randomly deletes my working code all day long. causing my big problems.

**Comment by M-Gonzalo** [comment]
This is a helping tool, not a whole devshop. If you're going to create a program, you NEED to know how to program. There is no way around it. No matter what the hypelords say, we're simply not there yet. We don't even know if we're ever going to get there

**Comment by paladyne-io** [comment]
> This is a helping tool, not a whole devshop. If you're going to create a program, you NEED to know how to program.  

I agree but I know how to program and it's still pretty annoying when Cline deletes working code (and comments) for no reason.  It's like a very helpful friend who sometimes goes behind your back to undermine you. :-) or something... It has  happened much more frequently in the past couple of days even though I have the following custom instructions:  
Never delete comments. 
Never delete existing methods or functions.

**Comment by saoudrizwan** [comment]
@stuinfla @paladyne-io @elogicuk I am sorry you guys are dealing with this. I anticipate it's happening more often than before because of the new Claude 3.5 Sonnet V2 model tending to be lazier than V1.

OpenAI just released a new API featured called [predicted outputs](https://platform.openai.com/docs/guides/latency-optimization#use-predicted-outputs) that would fix this issue entirely. It essentially spits out the tokens its read before and only has to "think" about new tokens (the new edits to the file), making the edit faster and more reliable without the code truncation. The issue is that this API is very rudimentary and limited (doesn't allow images, only works with gpt4o, etc) so it doesn't fit as nicely into Cline as I'd like. I'm hoping Anthropic comes out with their version of this API soon (which they've pre-released in Zed, called [Fast Edit Mode](https://www.youtube.com/watch?v=6CCuyTMhZKw)), so I can add it to Cline ASAP and fix this problem.

Pls join the [discord](https://discord.gg/cline) for updates about this, and thank you for bearing with me in the meantime 🙏

**Comment by M-Gonzalo** [comment]
> > This is a helping tool, not a whole devshop. If you're going to create a program, you NEED to know how to program.
> 
> I agree but I know how to program and it's still pretty annoying when Cline deletes working code (and comments) for no reason. It's like a very helpful friend who sometimes goes behind your back to undermine you. :-) or something... It has happened much more frequently in the past couple of days even though I have the following custom instructions: Never delete comments. Never delete existing methods or functions.

Yeah that's pretty annoying tho

**Comment by paladyne-io** [comment]
This is not very relevant to the original request*, which I agree with and also hope is improved but... just in case someone reads this, I have found some success by placing comments at the top of files with a description of what the component does and what to improve then adding  

PS. Claude 3.5 haiku was released recently and is now available in Cline Settings. It is more limited but approx. 1/3 of the cost.
